File: linked-list/3217.js

var modifiedList = function (nums, head) {
    const st = new Set(nums)
    let prev = null
    let curr = head
    while (curr !== null && st.has(curr.val)) {
        head = curr.next
        curr = head
    }
    while (curr != null) {
        const currVal = curr.val
        if (!st.has(currVal)) {
            prev = curr
            curr = curr.next
        } else {
            prev.next = curr.next
            curr = curr.next
        }
    }
    return head
};

File: linked-list/3217.py
class Solution:
    def modifiedList(
        self, nums: List[int], head: Optional[ListNode]
    ) -> Optional[ListNode]:
        st = set(nums)
        prev = None
        curr = head
        while curr is not None and curr.val in st:
            head = curr.next
            curr = head
        while curr is not None:
            currVal = curr.val
            if currVal not in st:
                prev = curr
                curr = curr.next
            else:
                prev.next = curr.next
                curr = curr.next
        return head
